### PR TITLE
Implemented a scroll-listener. Closes #47.

### LINKED ISSUE #47: Add listener for changing/scrolling month
I need a listener when scrolling through the calendar and the month changes.

Something like:

    mWeekView.setMonthScrolledListener(this);
    ...
    public void onMonthScrolled(int newYear, int newMonth) {
        // I would update some parts of my ui like the actionbar title with the current year and month (Google calendar wise)
    }

### PR DESCRIPTION
**Pull request by caske33**
The scroll listener emits an event every time the first visible day has changed.

(Closes #47 )

### CODE REVIEW

File: library/src/main/java/com/alamkanak/weekview/WeekView.java [hunk 1]
```diff
@@ -1418,6 +1430,17 @@ else if (hour * mHourHeight > mHourHeight * 24 - getHeight() + mHeaderTextHeight
...
+        /**
+         * Called when the first visible day has changed.
+         *
+         * (this will also be called during the first draw of the weekview)
+         * @param newFirstVisibleDay The new first visible day
+         * @param oldFirstVisibleDay The old first visible day (is null on the first call).
+         */
+        public void onFirstVisibleDayChanged(Calendar newFirstVisibleDay, Calendar oldFirstVisibleDay);
+    }
+
```
Comment (rachitmishra) [inline library/src/main/java/com/alamkanak/weekview/WeekView.java]: move interface to top

Comment (caske33) [inline library/src/main/java/com/alamkanak/weekview/WeekView.java]: Sorry but, this interface is in the section `interfaces`...

**Comment by CoderSpinoza** [comment]
@caske33 Any idean when this would be merged? :)

**Comment by caske33** [comment]
I can't tell you that. That's for @alamkanak to answer.

**Comment by caske33** [comment]
@CoderSpinoza it's merged :smile:

**Comment by Budincsevity** [comment]
Hi! I'm using the 1.2.1 version of the library, from gradle, but I don't see the setScrollListener() interface. It would be very useful for me. 

Besides that everything works perfect, I really love it.

**Comment by caske33** [comment]
`mWeekview.setScrollListener(listener)` doesn't work? (the listener is of type `WeekView.ScrollListener`)

**Comment by Budincsevity** [comment]
Yes, Android Studio shows "Cannot resolve method setScrollListener" error.

**Comment by caske33** [comment]
Apparently @alamkanak still has to release this to maven

**Comment by Shajeel-Afzal** [comment]
@caske33 how to add events in `onFirstVisibleDayChanged` callback? I mean i want to add events to show on WeekView, just like in `onMonthChange`. Thanks.

**Comment by caske33** [comment]
You can load new events in a cached list, call mWeekView.notifyDatasetChanged(), and then provide the new events when onMonthChange is called.

**Comment by Shajeel-Afzal** [comment]
How to access the cached list?

**Comment by caske33** [comment]
You cache this list for example in your fragment or activity (somewhere where your onMonthChange Listener can reach them). If you have data in this cached list, you return it, otherwise you fetch it for example from the network (or return an empty list, depending on what you want to acheive).

(PS if you do this, don't forget to use the parameters of the onMonthChange method, otherwise your events will show up 3 times!)

**Comment by piotrwojtowicz** [comment]
As question about getting data from network and cache appears quite often I prepared some template, which I am also successfully using at the moment. Any comments are appreciated. I have two flags to prevent WeekView from firing data loading three times. When I need to reload data in cache I can set both flags to false and call loadEvents again. 

    private List<someClass> _yourCache;

    private boolean _eventsLoading = false;
    private boolean _eventsLoaded = false;

    private void loadEvents() {

        //... load data for events into cache, i.e. load from network

        _eventsLoaded = true;
        weekview.notifyDatasetChanged();
    }

    private List<WeekViewEvent> GetEvents(int year, int month) {

        //... prepare events from data in cache
    }

    @Override
    public List<WeekViewEvent> onMonthChange(int newYear, int newMonth) {

        List<WeekViewEvent> result = new ArrayList<>();

        if (_eventsLoaded) {

            result = GetEvents(newYear, newMonth);

        } else if (!_eventsLoading) {

            _eventsLoading = true;
            loadEvents();
        }

        return result;
    }

**Comment by caske33** [comment]
I have something similair, although I use a cache for every month, and use that cache as my flag(s) at the same time

So the first time a month is requested, I return an empty list and store this list in an hashmap with a key that links to the month (and year). Every next call to onMonthChange will return this empty list.

When the network returns with data, I fill the list in the hashmap and call notifyDatasetChanged(). This will then return the data.

**Comment by Shajeel-Afzal** [comment]
@caske33 i am using day change listener. 'onFirstVisibleDayChanged' method is not being called when i change from August 2, 2015 to August 3, 2015. Any guess?

**Comment by caske33** [comment]
@Shajeel-Afzal Is that only those dates (so others work)? And is it always the same date? Or is the number of days difference with the current date that matters?

**Comment by vhenriques** [comment]
@caske33  ive implemented the WeekView.ScrollListener
but the onFirstVisibleDayChanged()
method is not getting fired when i scroll between days
in day view

**Comment by vhenriques** [comment]
crap it was my mistake
i didnt set the listener
 mWeekView.setScrollListener(getActivity());

**Pull request merged**